src/main/java/uk/gov/companieshouse/authcodenotification/controller/AuthCodeNotificationController.java

```diff
@@ -51,18 +57,35 @@ public ResponseEntity<Object> sendEmail(@RequestHeader(value = ERIC_REQUEST_ID_K
         var authCode = dataSanitiser.makeStringSafe(sendEmailRequestDto.getAuthCode());
 
         var logDataMap = new DataMap.Builder().companyNumber(companyNumber).build();
-        ApiLogger.infoContext(requestId,"Request received for auth code email", logDataMap.getLogMap());
+        var logMap = logDataMap.getLogMap();
+        ApiLogger.infoContext(requestId, "Request received for auth code email", logMap);
 
+        // validate request data
         var validationErrors =  authCodeEmailValidator.validate(companyNumber, authCode, new Errors(), requestId);
         if (validationErrors.hasErrors()) {
             ApiLogger.errorContext(requestId, String.format(VALIDATION_ERRORS_MESSAGE,
-                    convertErrorsToJsonString(validationErrors)), null, logDataMap.getLogMap());
+                    convertErrorsToJsonString(validationErrors)), null, logMap);
             var responseBody = ChResponseBody.createErrorsBody(validationErrors);
             return new ResponseEntity<>(responseBody, HttpStatus.BAD_REQUEST);
         }
 
+        // encrypt auth code
+        String authCodeEncrypted;
+        try {
+            authCodeEncrypted = encrypter.encrypt(authCode);
+            if (StringUtils.isBlank(authCodeEncrypted)) {
+                ApiLogger.errorContext(requestId, "Encrypted auth code is blank", null, logMap);
+                return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+            }
+            ApiLogger.infoContext(requestId, "Successfully encrypted auth code to: " + authCodeEncrypted , logMap);
+        } catch (Exception e) {
```

Comment:
Could you be more specific with the type of exception caught here? What can the `encrypter.encrypt()` call throw?

Reply:
Oh, I see, quite a few. Should we define a new specific EncryptionException that we catch, maybe?

Reply:
we can , but we'd then want to log the exception before it gets converted into EncryptionException. That means passing in requestId, logmap into the Encrypter. If the logging is in Encrypter then we won't be able to say what was being encrypted when it failed as we can't log the original value (for safety) so we would just have to say Encryption failed , but can't say what for. At least that's how I see it.

Reply:
Have decided not to log what the name of the thing being encrypted was and just rely on the stack trace, so am currently adding logging into the Encrypter to log before converting to a ServiceException (Tried EncryptionException but it didn't add much as it just gets converted to ServiceException to be thrown to the controller to generate the 500 response)

Reply:
I think here it is better to keep the log map out of the encrypter class so that it stays as a true util. So throw all the exceptions and catch them in the service where we then log.

Reply:
I think this can be resolved now. All agree (@pjames-ch @markpit)?